### PR TITLE
[vcpkg] adds progress bar

### PR DESCRIPTION
**Pull request by dan-shaw**
Adds a progress bar for downloading sources and CMake builds.

Adds option to disable progress bar output with  `--disable-progress-bar`

### CODE REVIEW

**Comment by PhoebeHui** [comment]
With this changes, currently, it looks like:

image

**Comment by PhoebeHui** [comment]
BTW, why it outputs 2 times 'Download Progress [..........] (100%)' for downloading msys in abvove feature?

**Comment by BillyONeal** [comment]
The qt failure seems to be broken due to improper dependency declaration of qt5-tools:

```
   Creating library ..\..\bin\qdistancefieldgenerator.lib and object ..\..\bin\qdistancefieldgenerator.exp
qsgd3d12backendd.lib(qsgd3d12engine.obj) : error LNK2019: unresolved external symbol CreateDXGIFactory2 referenced in function "private: void __cdecl QSGD3D12DeviceManager::ensureCreated(void)" (?ensureCreated@QSGD3D12DeviceManager@@AEAAXXZ)
Qt5Guid.lib(qrhid3d11.obj) : error LNK2001: unresolved external symbol IID_IDXGIFactory2
Qt5Guid.lib(qrhid3d11.obj) : error LNK2001: unresolved external symbol IID_ID3D11DeviceContext1
Qt5Guid.lib(qrhid3d11.obj) : error LNK2001: unresolved external symbol IID_ID3DUserDefinedAnnotation
..\..\bin\qdistancefieldgenerator.exe : fatal error LNK1120: 4 unresolved externals
```

@Neumann-A Do you have ideas here?

File: toolsrc/src/vcpkg/build.cpp [hunk 1]
```diff
@@ -691,10 +692,119 @@ namespace vcpkg::Build
...
         Checks::check_exit(VCPKG_LINE_INFO, out_file, "Failed to open '%s' for writing", fs::u8string(stdoutlog));
+        std::string outputStack;
+        bool downloadProgressMode = false;
+        bool buildProgressMode = false;
+
         const int return_code = System::cmd_execute_and_stream_data(
             command,
             [&](StringView sv) {
-                System::print2(sv);
+                if (action.build_options.disable_progress_bar == Build::ProgressBar::NO)
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/build.cpp:702]: This needs to be extracted and unit tested; it's very difficult to review without knowing the format it's attempting to parse in advance.

**Comment by BillyONeal** [comment]
Do you still want to merge this in September? (Asking since it's marked in our backlog that way)

**Comment by BillyONeal** [comment]
@dan-shaw it's been over 9 months on this PR and it doesn't merge cleanly; are you still working on it?

**Comment by PhoebeHui** [comment]
Closing PR due to inactivity. Please reopen if this is ongoing work.

**Pull request closed without merging**